### DISCUSSION TITLE
How can I get heic image size without decoding image?

### DISCUSSION THREAD

**Discussion by eeyrw** [bigcat88/pillow_heif#314] [Q&A] [answered]
Build millions images meta information database is quite common in deep learning. Image size is a kind of important meta info for cleaning dataset, but decoding heif is quite time consuming. So how can I use the pillow_heif read image size but do not decoding image?

**Comment by bigcat88** [comment]
just read `size` property and do not read `data` property or anything that will lead to image decoding.

**Reply by eeyrw** [reply]
Something like this?
```
    heif_file = pillow_heif.open_heif(file, convert_hdr_to_8bit=False)
    print("Number of images:", len(heif_file))
    print("Information about each image:")
    for image in heif_file:
        print("\tSize:", image.size)
```

**Reply by bigcat88** [reply, marked as answer]
Yes, this will not decode the image, just take the `size` information from image header.
